doc/whats_new/v1.0.rst

```diff
@@ -257,6 +257,10 @@ Changelog
   :class:`ensemble.StackingClassifier` and :class:`ensemble.StackingRegressor`.
   :pr:`19564` by `Thomas Fan`_.
 
+- |Fix| Fixed the range of the argument max_samples to be (0.0, 1.0]
+  in :class:`ensemble.BaseForest` and the class that inherits it.
```

Comment:
`BaseForest` is not really part of the public API. Maybe we could just mention something like

 :class:`ensemble.RandomForestClassifier`,  :class:`ensemble.RandomForestRegressor` and all other random forest based estimators.

Reply:
fixed it on https://github.com/scikit-learn/scikit-learn/pull/20159/commits/2ce6f58f946810c74f101de7ea741fb80e56ccc8